src/Test.FunctionalTests.BinSkim.Rules/RuleTests.cs

```diff
@@ -277,13 +278,23 @@ private BinaryAnalyzerContext CreateContext(TestMessageLogger logger, Properties
                 context.Rule = skimmer;
 
                 AnalysisApplicability applicability = skimmer.CanAnalyze(context, out string reasonForNotAnalyzing);
+
                 if (applicability != expectedApplicability)
                 {
                     sb.AppendLine("CanAnalyze did not correct indicate target applicability (unexpected return was " +
                         applicability + "): " +
                         Path.GetFileName(target));
                     continue;
                 }
+
+                if (expectedReasonForNotAnalyzing != null && reasonForNotAnalyzing != expectedReasonForNotAnalyzing)
+                {
+                    sb.AppendLine("Cannot Analyze but unexpected reason identified (unexpected return was " +
+                        reasonForNotAnalyzing + "but " +
+                        expectedReasonForNotAnalyzing + "was expected): " +
```

Comment:
>was [](http://example.com/codeflow?start=57&length=3)

Looks like you are missing a space at the beginning of this literal. This raises a very common, more advanced piece of guidance I give all engineers: you should always provoke and examine the output of your test failures to find bugs like this and ensure that your new testing is providing a good experience in failure conditions (i.e., actionable for the poor soul who is responsible for fixing the future break).

It's great that you've added this code path by the way!!